src/server/game/Battlegrounds/BattlegroundQueue.cpp

```diff
@@ -712,6 +712,11 @@ void BattlegroundQueue::BattlegroundQueueUpdate(BattlegroundBracketId bracket_id
                 for (auto itr : m_SelectionPools[TEAM_ALLIANCE + i].SelectedGroups)
                     BattlegroundMgr::InviteGroupToBG(itr, bg, itr->RealTeamID);
         }
+
+        if (!bg_template->isArena() && !bgsToCheck.empty()) {
```

Comment:
just for me to understand... What does 
```
!bgsToCheck.empty()
```
do?

Reply:
It checks if there are non-empty BGs

Reply:
I guess why next question would be, why would there be an empty BG? Or is that the "phase" when nobody as accepted the BG queue?

Reply:
@Kitzunu sorry I wasn't clear enough:

I check if there are still BGs that are running AND that are NOT full. Under this condition, there is no reason to go further and start new BGs.

New BGs should be started when there are NO non-full BGs (so either when there are no BGs at all or when all the current BGs are full).

I hope it's clear now. If it's not clear yet, please let me know :)

Reply:
Yea, makes sense.

I have not tested the PR yet, but how would this work if for example:

Alliance Team: 4/5
Horde Team: 4/5

In Queue: Two groups of two (2 Ally 2horde both in a party)

Would it start a new game, or just sit endlessly in a queue until the 2 last spots in the first game is filled?

Reply:
I haven't tested such a scenario, however this should not be affected by my PR

Reply:
The bug is inside "CheckNormalMatch". I'll explain you it in PVT